src/vs/workbench/api/common/notebookSearchExtHost.ts

```diff
@@ -3,111 +3,38 @@
-export interface IFileMatchWithCells extends IFileMatch {
-	cellResults: ICellMatch[];
-}
-
-export interface ICellMatch {
-	cell: ICellViewModel | CellSearchModel;
-	index: number;
-	contentResults: ITextSearchMatch[];
-	webviewResults: ITextSearchMatch[];
-}
-export function isIFileMatchWithCells(object: IFileMatch): object is IFileMatchWithCells {
-	return 'cellResults' in object;
-}
-
-// to text search results
-
-export function contentMatchesToTextSearchMatches(contentMatches: FindMatch[], cell: ICellViewModel | CellSearchModel): ITextSearchMatch[] {
-	return genericCellMatchesToTextSearchMatches(
-		contentMatches,
-		cell instanceof CellSearchModel ? cell.inputTextBuffer : cell.textBuffer,
-		cell
-	);
-}
-
-export function genericCellMatchesToTextSearchMatches(contentMatches: FindMatch[], buffer: IReadonlyTextBuffer, cell: ICellViewModel | CellSearchModel) {
-	let previousEndLine = -1;
-	const contextGroupings: FindMatch[][] = [];
-	let currentContextGrouping: FindMatch[] = [];
-
-	contentMatches.forEach((match) => {
-		if (match.range.startLineNumber !== previousEndLine) {
-			if (currentContextGrouping.length > 0) {
-				contextGroupings.push([...currentContextGrouping]);
-				currentContextGrouping = [];
-			}
-		}
-
-		currentContextGrouping.push(match);
-		previousEndLine = match.range.endLineNumber;
-	});
-
-	if (currentContextGrouping.length > 0) {
-		contextGroupings.push([...currentContextGrouping]);
-	}
-
-	const textSearchResults = contextGroupings.map((grouping) => {
-		const lineTexts: string[] = [];
-		const firstLine = grouping[0].range.startLineNumber;
-		const lastLine = grouping[grouping.length - 1].range.endLineNumber;
-		for (let i = firstLine; i <= lastLine; i++) {
-			lineTexts.push(buffer.getLineContent(i));
-		}
-		return new TextSearchMatch(
-			lineTexts.join('\n') + '\n',
-			grouping.map(m => new Range(m.range.startLineNumber - 1, m.range.startColumn - 1, m.range.endLineNumber - 1, m.range.endColumn - 1)),
-		);
-	});
-
-	return textSearchResults;
-}
-
-export function webviewMatchesToTextSearchMatches(webviewMatches: CellWebviewFindMatch[]): ITextSearchMatch[] {
-	return webviewMatches
-		.map(rawMatch =>
-			(rawMatch.searchPreviewInfo) ?
-				new TextSearchMatch(
-					rawMatch.searchPreviewInfo.line,
-					new Range(0, rawMatch.searchPreviewInfo.range.start, 0, rawMatch.searchPreviewInfo.range.end),
-					undefined,
-					rawMatch.index) : undefined
-		).filter((e): e is ITextSearchMatch => !!e);
-}
-
-// experimental
 
 interface RawOutputFindMatch {
 	textBuffer: IReadonlyTextBuffer;
 	matches: FindMatch[];
 }
 
-export const rawCellPrefix = 'rawCell#';
-export class CellSearchModel extends Disposable {
+export interface ICellSearchModel {
+	inputTextBuffer: IReadonlyTextBuffer;
+	outputTextBuffers: IReadonlyTextBuffer[];
+	findInInputs(target: string): FindMatch[];
+	findInOutputs(target: string): RawOutputFindMatch[];
+}
+
+export class CellSearchModel extends Disposable implements ICellSearchModel {
```

Comment:
I'm wondering if this piece can be moved into `workbench/contrib/notebook/common` since this runs on `textModel` and it's very generic.

Reply:
I could probably just move it to workbench/contrib/search/common since a lot of notebook search stuff is there already.